sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -2246,6 +2246,10 @@ pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
   bool DeviceIsIntegrated = Context->Devices.size() == 1 &&
                             Context->Devices[0]->ZeDeviceProperties.flags &
                                 ZE_DEVICE_PROPERTY_FLAG_INTEGRATED;
+  if (DeviceIsIntegrated)
+    zePrint("Buffer Create: Integrated GPU will use zeMemAllocHost\n");
+  else
+    zePrint("Buffer Create: Discrete GPU will use zeMemAllocDevice\n");
```

Comment:
I don't think we need to pollute the traces with these pritns. Note that ZE_CALL macro already prints the L0 calls being made, just use those.

Reply:
From the existing L0 calls we cannot tell if discrete or integrated GPU is detected.

Reply:
you can query the device if it is integrated or not, from the sycl program